src/basic/log.c

```diff
@@ -347,18 +347,24 @@ static int write_to_console(
                 iovec[n++] = IOVEC_MAKE_STRING(prefix);
         }
 
-        highlight = LOG_PRI(level) <= LOG_ERR && show_color;
+        if (show_color) {
+                on =    LOG_PRI(level) <= LOG_ERR ? ANSI_HIGHLIGHT_RED :
+                        LOG_PRI(level) <= LOG_WARNING ? ANSI_HIGHLIGHT_YELLOW :
```

Comment:
let's make this match journalctl's output i.e.:

```c
on = LOG_PRI(level) <= LOG_ERR ? ANSI_HIGHLIGHT_RED : 
        LOG_PRI(level) <= LOG_NOTICE ? ANSI_HIGHTLIGHT : 
        LOG_PRI(level) >= LOG_DEBUG ? ANSI_GREY :
        NULL;
```

In fact, that should probably be moved into a set of functions in syslog-util.h or so, so that this can be shared between journalctl and log.c